types/prop-types/index.d.ts

```diff
@@ -5,11 +5,15 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
+export type ReactComponentLike =
+    | string
+    | ((props: any, context?: any) => null)
+    | (new (props: any, context?: any) => null);
+
 export interface ReactElementLike {
-    type: string | ((...args: any[]) => ReactElementLike);
+    type: ReactComponentLike;
```

Comment:
The original `type` property for this was not exactly the same as the one in the React type definitions, causing issues with the `PropTypes.element` validator, which this fixes.